docs/README-ock.rst

```diff
@@ -0,0 +1,123 @@
+Opencryptoki
+============
+
+The PKCS#11 implementation `opencryptoki` can be used with the
+pkcs11-provider. This document describes a basic installation and setup for
+testing purposes with the pkcs11-provider.
+
+Installation as package
+=======================
+
+Install opencryptoki from the distribution's repository.
+
+RedHat/Fedora
+-------------
+
+Install the base packages (including depending packages):
+
+    sudo dnf install opencryptoki
+
+For the pkcs11-provider testing environment, the software token is required.
+Other token libraries can be installed as well.
+
+    sudo dnf install opencryptoki-swtok
+
+The location of the default configuration for opencryptoki is
+`/etc/opencryptoki/opencryptoki.conf`
+
+Debian/Ubuntu
+-------------
+
+Install the base packages (including depending packages):
+
+    sudo apt install opencryptoki
+
+The location of the default configuration for opencryptoki is
+`/etc/opencryptoki/opencryptoki.conf`
+
+Installation from source
+========================
+
+Get the source from github, build and install.
+		
+    git clone https://github.com/opencryptoki/opencryptoki.git
+    cd opencryptoki
+    ./bootstrap
+    ./configure
+    make
+    sudo make install
+    sudo ldconfig
+
+The location of the default configuration for opencryptoki is
+`/usr/local/etc/opencryptoki/opencryptoki.conf`
+
+Configuration
+=============
+
+For the pkcs11-provider tests, the default configuration should be
+sufficient. If the opencryptoki configuration has changed, please run the
+initialization script once.
+
+    sudo scripts/ock-once-swtok.sh
```

Comment:
Please find a way to run these commands as a normal user,
I really do not want a test suite that requires root.

If it is not possible to use env vars to tell opencryptoki which config files to use (so they can be placed in a temp dir and not pollute the distro environment) I will punt on having official cryptoki tests.

See the softoken tests, no root is required, same should be here.

Reply:
At the moment, it is not possible to even run pkcsslotd without being root. This restriction may be removed in the future, but we're not yet there. I can keep the tests in my fork until then.